tests/signers.py

```diff
@@ -42,20 +45,44 @@ def __init__(self, private_key):
     async def send_transaction(self, account, to, selector_name, calldata, nonce=None, max_fee=0):
         return await self.send_transactions(account, [(to, selector_name, calldata)], nonce, max_fee)
 
-    async def send_transactions(self, account, calls, nonce=None, max_fee=0):
-        if nonce is None:
-            execution_info = await account.get_nonce().call()
-            nonce, = execution_info.result
-
+    async def send_transactions(
+        self,
+        account,
+        calls,
+        nonce=None,
+        max_fee=0
+    ) -> TransactionExecutionInfo:
+        # hexify address before passing to from_call_to_call_array
         build_calls = []
         for call in calls:
             build_call = list(call)
             build_call[0] = hex(build_call[0])
             build_calls.append(build_call)
 
-        (call_array, calldata, sig_r, sig_s) = self.signer.sign_transaction(
-            hex(account.contract_address), build_calls, nonce, max_fee)
-        return await account.__execute__(call_array, calldata, nonce).invoke(signature=[sig_r, sig_s])
+        raw_invocation = get_raw_invoke(account, build_calls)
+        state = raw_invocation.state
+
+        if nonce is None:
+            nonce = await state.state.get_nonce_at(account.contract_address)
+
+        _, sig_r, sig_s = self.signer.sign_transaction(int(account.contract_address), build_calls, nonce, max_fee)
```

Comment:
```suggestion
        _, sig_r, sig_s = self.signer.sign_transaction(account.contract_address, build_calls, nonce, max_fee)
```
this is no longer needed with the new nile signer :)